### PR TITLE
Add Chalmet algorithm

### LINKED ISSUE #15: Algorithms to add
- [ ] Chalmet #41 

X-ref the implementation in vOptGeneric: 
https://github.com/vOptSolver/vOptGeneric.jl/blob/5cdecf89d6fe180bda99b8f20c92b576b13d1db2/src/algorithms.jl#L248-L366

- [x] Dichotomy #16

x-ref the implementation in vOptGeneric: 
https://github.com/vOptSolver/vOptGeneric.jl/blob/5cdecf89d6fe180bda99b8f20c92b576b13d1db2/src/algorithms.jl#L143-L246

### PR DESCRIPTION
**Pull request by kofgokhan**
Link for the algorithm: https://doi.org/10.1016/0377-2217(86)90093-7

I looked at NISE.jl implementation. Maybe a few more functions can be added to make `optimize_multiobjective!` more readable although I was not sure how to set it up.

Also, I did not add any attributes to this one either.

Closes #15

### CODE REVIEW

**Comment by odow** [comment]
> Maybe a few more functions can be added to make optimize_multiobjective! more readable although I was not sure how to set it up.

Looks okay for now. We can always improve later.

> Also, I did not add any attributes to this one either.

Can improve later.

File: src/algorithms/Chalmet.jl [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    X, Y = _compute_point(model, variables, model.f)
+    return status, SolutionPoint(X, Y)
+end
+
+function optimize_multiobjective!(algorithm::Chalmet, model::Optimizer)
+    if MOI.output_dimension(model.f) != 2
+        error("Chalmet requires exactly two objectives")
+    end
+    if MOI.output_dimension(model.f) == 1
+    end
```
Comment (odow) [inline src/algorithms/Chalmet.jl:40]: Delete this for now. It's okay to require exactly two objectives

File: test/algorithms/Chalmet.jl [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+import HiGHS
+import MultiObjectiveAlgorithms as MOA
+
+const MOI = MOA.MOI
+
+function run_tests()
+    if Sys.WORD_SIZE == 32
+        return  # Skip on 32-bit because HiGHS fails
+    end
```
Comment (odow) [inline test/algorithms/Chalmet.jl:18]: Is this needed? Ideally, we should only skip tests if HiGHS actually fails

File: src/algorithms/Chalmet.jl [hunk 3]
```diff
@@ -0,0 +1,105 @@
+#  Copyright 2019, Oscar Dowson and contributors
+#  This Source Code Form is subject to the terms of the Mozilla Public License,
+#  v.2.0. If a copy of the MPL was not distributed with this file, You can
+#  obtain one at http://mozilla.org/MPL/2.0/.
+
+"""
+    Chalmet()
+`Chalmet` implements the algorithm of:
```
Comment (odow) [inline src/algorithms/Chalmet.jl:8]:
```suggestion

`Chalmet` implements the algorithm of:
```

File: src/algorithms/Chalmet.jl [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+...
```
Comment (odow) [inline src/algorithms/Chalmet.jl:9]:
```suggestion

Chalmet, L.G., and Lemonidis, L., and Elzinga, D.J. (1986). An algorithm for the
bi-criterion integer programming problem. European Journal of Operational
Research. 25(2), 292-300
```

**Comment by odow** [comment]
The README also needs updating

File: src/algorithms/Chalmet.jl [hunk 5]
```diff
@@ -0,0 +1,108 @@
+#  Copyright 2019, Oscar Dowson and contributors
+#  This Source Code Form is subject to the terms of the Mozilla Public License,
+#  v.2.0. If a copy of the MPL was not distributed with this file, You can
+#  obtain one at http://mozilla.org/MPL/2.0/.
+
+"""
+    Chalmet()
+
```
Comment (odow) [inline src/algorithms/Chalmet.jl:8]:
```suggestion
```

**Review by odow — APPROVED** [review]

**Comment by odow** [comment]
Looks good! Once this is merged I think this package is in a good place. I'm interested to see what feedback we get once people start using it

**Comment by xgandibleux** [comment]
Indeed!!! It is crazy to see the speed how this package has been developped these last weeks. I will contact the authors of  algorithms recently published that I know personally  (N. Boland, D. Vanderpooten, A. Lohne, etc.) to invite them to add their implementation. Same for us (we have several codes that we could add, after a refactoring/reimplementation from C to Julia). I have also some ideas to facilitate the querying of results. And I am exited to present this new package in my lectures

**Pull request merged**